tests/backends/console/test_env.py

```diff
@@ -54,3 +54,31 @@ def mock_input(prompt: str) -> str:
 
     console = Console("TestBoard", input_function=mock_input)
     assert console.read("Enter test input", None) is None
+
+
+BAD_ATTEMPT_COUNT = 0
```

Comment:
I think it would be better practice to move `mock_input` and `mock_print` into a class, make `BAD_ATTEMPT_COUNT` an attribute of the class, and pass the functions into `Console()` as bound methods.

```python
def test_console_read_bad_type() -> None:
  class Mock:
    def __init__(self):
      self.bad_attempt_count = 0
    def input(self, prompt: str) -> str:
      ...
    def print(self, text: str) -> None:
      ...
  mock = Mock()
  console = Console(
    "TestConsole",
    print_function=mock.print,
    input_function=mock.input,
  )
  ...
```

This pattern is strongly suggesting to me that the `Console` class isn't very pythonic - changing functionality should be done by overriding in a subclass, not by passing functions into its constructor.